pkg/innerring/state.go

```diff
@@ -104,9 +106,21 @@ func (s *Server) voteForSidechainValidator(validators keys.PublicKeys) error {
 
 	epoch := s.EpochCounter()
 
+	var (
+		nonce uint32 = 1
+		vub   *uint32
+		err   error
+	)
+
+	if prm.Hash != nil {
+		nonce, vub, err = s.morphClient.CalculateNonceAndVUB(*prm.Hash)
+		if err != nil {
+			return fmt.Errorf("could not calculate nonce and `validUntilBlock` values: %w", err)
+		}
+	}
```

Comment:
Maybe some methods do not require `NonceAndVUB` calculations like this one. But we can keep it for similarity with others notary invocations.

Reply:
Yes, you are right. I thought that epoch number is enough to keep transactions unique, but Alphabet catches notifications from role management contract, so there can be multiple votes per epoch.